device_backends/Subdevice/src/SubdeviceRegisterAccessor.cc

```diff
@@ -26,9 +26,11 @@ namespace ChimeraTK {
 
   bool SubdeviceRegisterAccessor::doWriteTransfer(ChimeraTK::VersionNumber) {
     std::lock_guard<decltype(_backend->mutex)> lockGuard(_backend->mutex);
+    // This is "_numberOfWords / _accData->getNumberOfSamples()" rounded up:
+    auto nTransfers = (_numberOfWords + _accData->getNumberOfSamples() - 1) / _accData->getNumberOfSamples();
     try {
       size_t idx = 0;
-      for(size_t adr = _byteOffset; adr < _byteOffset + 4 * _numberOfWords; adr += 4) {
+      for(size_t adr = _startAddress; adr < _startAddress + nTransfers; ++adr) {
```

Comment:
Having more than one transfer and increasing the address contradicts the comment in the ticket that address spaces can be non-contiguous (for instance if they are bit patterns)

Reply:
True, but I did not want to break this existing feature at this time. Since this depends only on information present in the map file, one simply needs to pay attention to define registers which are not longer than the target data register when the address space is non-contiguous.